activation/poet_client_test.go

```diff
@@ -525,3 +532,123 @@ func TestPoetService_CachesPowParams(t *testing.T) {
 		})
 	}
 }
+
+func TestPoetService_FetchPoetPhaseShift(t *testing.T) {
+	t.Parallel()
+	const phaseShift = time.Second
+
+	t.Run("poet service created: expected and fetched phase shift are matching",
+		func(t *testing.T) {
+			cfg := DefaultPoetConfig()
+			cfg.PhaseShift = phaseShift
+
+			client := NewMockPoetClient(gomock.NewController(t))
+			client.EXPECT().Address().Return("some_addr").AnyTimes()
+			client.EXPECT().Info(gomock.Any()).Return(&types.PoetInfo{
+				PhaseShift: phaseShift,
+			}, nil)
+
+			NewPoetServiceWithClient(nil, client, cfg, zaptest.NewLogger(t))
+		})
+
+	t.Run("poet service created: phase shift is not fetched",
+		func(t *testing.T) {
+			cfg := DefaultPoetConfig()
+			cfg.PhaseShift = phaseShift
+
+			client := NewMockPoetClient(gomock.NewController(t))
+			client.EXPECT().Address().Return("some_addr").AnyTimes()
+			client.EXPECT().Info(gomock.Any()).Return(nil, errors.New("some error"))
+
+			NewPoetServiceWithClient(nil, client, cfg, zaptest.NewLogger(t))
+		})
+
+	t.Run("poet service creation failed: expected and fetched phase shift are not matching",
+		func(t *testing.T) {
+			cfg := DefaultPoetConfig()
+			cfg.PhaseShift = phaseShift
+
+			client := NewMockPoetClient(gomock.NewController(t))
+			client.EXPECT().Address().Return("some_addr").AnyTimes()
+			client.EXPECT().Info(gomock.Any()).Return(&types.PoetInfo{
+				PhaseShift: phaseShift * 2,
+			}, nil)
+
+			defer func() {
+				if r := recover(); r == nil {
+					t.Errorf("Panic is expected")
+				}
+			}()
```

Comment:
If you use `log.Fatal` instead of `log.Panic` you can use a hook in tests to assert that `log.Fatal` was called. As an example you can take a look at the `fatalHook` in `activation/post_supervisor_test.go`

Reply:
It is same as here
https://github.com/spacemeshos/go-spacemesh/pull/6116#discussion_r1701432124

Reply:
Yes, but you used `defer` and `recover`?

Reply:
sure, but I thought we want to use Fatal instead of panic anyway (because of the stacktrace in the message)